posthog/models/feedback/survey.py

```diff
@@ -267,10 +270,78 @@ def update_response_sampling_limits(sender, instance, **kwargs):
     instance.response_sampling_daily_limits = json.dumps(schedule)
 
 
+def update_question_stable_indices(sender, instance: Survey):
+    """
+    Ensures all questions have a stable_index and maintains max_question_stable_index.
+    This runs before saving a survey, ensuring indices are always computed server-side.
+    Key behaviors:
+    - New surveys: Assigns sequential indices starting from 0
+    - Existing surveys:
+        - Preserves stable_index for existing questions
+        - Assigns new indices only to questions without them
+        - Maintains max_question_stable_index as the highest index ever used
+    - Editing surveys:
+        - Keeps existing stable_index values even when questions are reordered
+        - When adding new questions, uses max_question_stable_index + 1
+    IMPORTANT: stable_index values must be unique across all questions as they're
+    used to track survey responses in the backend.
+    """
+    if not instance.questions:
+        instance.max_question_stable_index = -1
+        return
+
+    # If this is a new survey (no ID yet), start from 0
+    if not instance.id:
+        for i, question in enumerate(instance.questions):
+            question["stable_index"] = i
+        instance.max_question_stable_index = len(instance.questions) - 1
+        return
+
+    # For existing surveys, load the current state from DB
+    try:
+        current_survey = Survey.objects.get(id=instance.id)
+        current_max_index = current_survey.max_question_stable_index or -1
+    except Survey.DoesNotExist:
+        current_max_index = -1
```

Comment:
exactly what I said [here](https://github.com/PostHog/posthog/pull/29114/files#r1969287408)

this code isn't synchronized for is running within a database transaction, so you will get bugs when the same survey is edited at the same time